src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -5851,6 +5851,15 @@ protected override bool IsItemContainer(string path)
                         destination = MakePath(destination, dir.Name);
                     }
 
+                    // Don't allow moving a directory into itself
+                    if (destination.StartsWith(Path.TrimEndingDirectorySeparator(path) + Path.DirectorySeparatorChar))
```

Comment:
Uh, as discussed previously this addresses only the common scenario but not possible DFS, mount point and symlink scenarios. I suggest to discuss the problem with .Net team how distinguish these cases since we always get IOException with the same HResult -2146232800 (0x80131620).